### ISSUE TITLE
JDK 17

### ISSUE THREAD

**Issue by 13lipca** [threadly/threadly#272]
Do you have any plans (if yes when) to support threadly on JDK 17?

**Comment by jentfoo** [comment]
Thank you for filing an issue @13lipca

Yes we do plan on supporting this project with JDK 17.  The notable work that needs to completed is issue #256 (other needed changes are straight forward).  There was some internal discussion about exactly what this implementation should look like, but that has since been concluded and I have started development on this.  After #256 is completed the plan is to release version 7.0 which will drop support for older JDK versions while simultaneously bringing in the needed support for JDK 17.  

Time has certainly been a constraint for me lately, and so development has slowed, but the project is still actively supported and this is something I want to complete soon.  Is there a timeline that you need this to be completed by in order to facilitate your projects needs?

**Comment by 13lipca** [comment]
Big thank you for the fast response and great work on the lib. We are currently migrating from JDK 15 to 17 - as you probably know 15 support ended on 31 Mar 2021.  We are releasing our software on the monthly basis, so the next release which we are planing is at the end of Jan, then another at end of Feb. If we could upgrade threadly 2 weeks before our release to have full quantity of the software stability would be nice. Long story short - mid of Feb sounds perfect. Thank you.

**Comment by jentfoo** [comment]
@13lipca, Since this work has already started I think that timeline is completely reasonable.  I will try to target a threadly 7.0 release with JDK 17.0 support before February.  That way there is some additional buffer as well as time for us to address any unexpected consequences before the mid February timeline.  Thank you again for the issue!

**Comment by jentfoo** [comment]
@13lipca We have released a `7.0-SNAPSHOT` artifact which is JDK 17 compatible.  If you want feel free to review the PR for the upcoming 7.0 release: https://github.com/threadly/threadly/pull/276

@13lipca if you have time we would appreciate any testing you can do on the 7.0-SNAPSHOT artifact before we release it (Pretty typical sonatype setup, but feel free to read these docs https://github.com/threadly/threadly/wiki/Getting-the-latest-snapshot-artifacts).  I am planning to release 7.0 by this weekend right now.  Thank you!

**Comment by 13lipca** [comment]
Hi @jentfoo,  we did roughly some testing and looks promising, so far our system starts correctly and also works correctly. We are working on high-performance solution with multithreading involved so we will monitor how it looks on our Test environments in the next days. In case of any issues will be found I will let you know. Thank you very much for a quick solution.

**Comment by jentfoo** [comment]
Thank you for the verification @13lipca.  I will probably release 7.0 on Saturday.  If you find any unexpected behaviors, or want help in optimizing, feel free to open a new issue.  I am happy to help with methods for efficient threadly usage

**Comment by jentfoo** [comment]
7.0 is merged and will be available in Maven Central shortly.  Thank you again for the issue @13lipca!
